main/inc/ajax/admin.ajax.php

```diff
@@ -35,6 +35,74 @@
     case 'version':
         echo version_check();
         break;
+
+    case 'save_block_extra':
+        $content = isset($_POST['content']) ? Security::remove_XSS($_POST['content']) : null;
+        $blockId = isset($_POST['block']) ? Security::remove_XSS($_POST['block']) : null;
+
+        if (empty($blockId)) {
+            die;
+        }
+
+        if (api_is_multiple_url_enabled()) {
+            $accessUrlId = api_get_current_access_url_id();
+
+            if ($accessUrlId == -1) {
+                die;
+            }
+
+            $urlInfo = api_get_access_url($accessUrlId);
+            $url = api_remove_trailing_slash(preg_replace('/https?:\/\//i', '', $urlInfo['url']));
+            $cleanUrl = replace_dangerous_char($url);
+            $cleanUrl = str_replace('/', '-', $cleanUrl);
+
+            $newUrlDir = api_get_path(SYS_PATH) . "home/$cleanUrl/admin/";
+        } else {
+            $newUrlDir = api_get_path(SYS_PATH) . "home/admin/";
+        }
+
+        if (!is_dir($newUrlDir)) {
+            mkdir($newUrlDir, api_get_permissions_for_new_directories(), true);
```

Comment:
If making a system call, you should always choose between two possibilities: 1) recover the return value in a variable and suffocate the error (with "@"), or 2) manage exceptions. Either way, the current mkdir(), unlink(), touch() and file_put_contents() calls are all dangerous as they could fail and show too much information to potential attackers.